src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java

```diff
@@ -340,9 +339,8 @@ public HttpResponse send(HttpRequest request, HttpRequestOptions options) throws
    * will block waiting to allocate them. Likewise, read/write speed differences could cause issues. The buffer size can be
    * customized for these reason.
    */
-  private HttpResponse sendAndDefer(HttpRequest request, HttpRequestOptions options, int currentRedirects)
+  private HttpResponse sendAndDefer(HttpRequest request, Request grizzlyRequest, HttpRequestOptions options, int currentRedirects)
```

Comment:
Why is the need for a request and grizzlyRequest. Add documentation indicating why this is needed.

Reply:
Because I made this method recursive. In case the server response is a 30x we create a new request in line 357 and call this method again.